minidump-processor/src/processor.rs

```diff
@@ -93,17 +93,7 @@ where
         .get_stream::<MinidumpSystemInfo>()
         .or(Err(ProcessError::MissingSystemInfo))?;
 
-    let os_version = format!(
-        "{}.{}.{}",
-        dump_system_info.raw.major_version,
-        dump_system_info.raw.minor_version,
-        dump_system_info.raw.build_number
-    );
-
-    let os_build = dump_system_info
-        .csd_version()
-        .map(|v| v.trim().to_owned())
-        .filter(|v| !v.is_empty());
+    let (os_version, os_build) = dump_system_info.os_parts();
```

Comment:
your approach and reasoning seems sensible to me.

Reply:
Without reading your entire patch: I think it's important that the `Minidump*` classes provide access to the exact data contained within the minidump, because there are lots of ways to interpret specific pieces of data. However, it's totally fine to also have helper methods to pick out specific useful bits that require interpretation, such as the `code_id` / `debug_id` bits on `MinidumpModule`.

Reply:
Thank you for the context! In that case, I'll stick with what's in the PR right now as it respects that and I'll add just a little bit extra to the helper's docstring to provide some guidance on what it does.